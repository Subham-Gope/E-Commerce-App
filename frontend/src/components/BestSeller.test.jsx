// import { describe, expect, test } from "vitest";
// import { screen, render } from "@testing-library/react";
// import ShopContext from "../context/ShopContext";
// import BestSeller from "./BestSeller";
// import { MemoryRouter } from "react-router-dom";

// // Mock products
// const mockProducts = [
//   {
//     _id: "1",
//     name: "Product 1",
//     image: ["img1.png"],
//     price: 100,
//     bestseller: true,
//   },
//   {
//     _id: "2",
//     name: "Product 2",
//     image: ["img2.png"],
//     price: 200,
//     bestseller: true,
//   },
//   {
//     _id: "3",
//     name: "Product 3",
//     image: ["img3.jpg"],
//     price: 300,
//     bestseller: true,
//   },
//   {
//     _id: "4",
//     name: "Product 4",
//     image: ["img4.jpg"],
//     price: 400,
//     bestseller: true,
//   },
//   {
//     _id: "5",
//     name: "Product 5",
//     image: ["img5.jpg"],
//     price: 500,
//     bestseller: true,
//   },
//   {
//     _id: "6",
//     name: "Product 6",
//     image: ["img6.jpg"],
//     price: 600,
//     bestseller: false,
//   }, // this product shall no appear in the render
// ];

// // Resuable render function
// const renderWithContext = (ui, value) => {
//   return render(
//     <MemoryRouter>
//       <ShopContext.Provider value={value}>{ui}</ShopContext.Provider>
//     </MemoryRouter>
//   );
// };

// describe("Testing the BestSeller.tsx file", () => {
//   test("renders only 5 bestseller items", () => {
//     // rendering the component
//     renderWithContext(<BestSeller />, { products: mockProducts });

//     // Checking the title texts
//     const title = screen.getByText(/best seller/i);
//     expect(title).toBeInTheDocument();
//     const description = screen.getByText(/shop our best sellers/i);
//     expect(description).toBeInTheDocument();

//     // first products shall be in the document
//     const product1 = screen.getByText(/product 1/i);
//     expect(product1).toBeInTheDocument();
//     const product2 = screen.getByText(/product 2/i);
//     expect(product2).toBeInTheDocument();
//     const product3 = screen.getByText(/product 3/i);
//     expect(product3).toBeInTheDocument();
//     const product4 = screen.getByText(/product 4/i);
//     expect(product4).toBeInTheDocument();
//     const product5 = screen.getByText(/product 5/i);
//     expect(product5).toBeInTheDocument();

//     // expect not to be in the document
//     const product6 = screen.queryByText(/product 6/i);
//     expect(product6).toBeNull();
//   });
// });
