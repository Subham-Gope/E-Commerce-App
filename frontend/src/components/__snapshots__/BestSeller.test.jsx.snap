// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the BestSeller.tsx file > renders only 5 bestseller items 1`] = `
<div>
  <div
    class="my-10"
  >
    <div
      class="text-center text-3xl py-8"
    >
      <div
        class="inline-flex gap-2 items-center mb-3"
      >
        <p
          class="text-gray-500"
        >
          BEST
           
          <span
            class="text-gray-700 font-medium"
          >
            SELLER
          </span>
        </p>
        <p
          class="w-8 sm:w-12 h-[1px] sm:h-[2px] bg-gray-700"
        />
      </div>
      <p
        class="w-3/4 m-auto text-xs sm:text-sm md:text-base text-gray-600"
      >
        Shop Our Best Sellers: Customer Favorites Youâ€™ll Love!
      </p>
    </div>
    <div
      class="grid grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5 gap-4 gap-y-6"
    >
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/1"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="img1.png"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          Product 1
        </p>
        <p
          class="text-sm font-medium"
        >
          100
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/2"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="img2.png"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          Product 2
        </p>
        <p
          class="text-sm font-medium"
        >
          200
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/3"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="img3.jpg"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          Product 3
        </p>
        <p
          class="text-sm font-medium"
        >
          300
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/4"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="img4.jpg"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          Product 4
        </p>
        <p
          class="text-sm font-medium"
        >
          400
        </p>
      </a>
      <a
        class="text-gray-700 cursor-pointer"
        data-discover="true"
        href="/product/5"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="product image"
            class="hover:scale-110 transition ease-in-out"
            src="img5.jpg"
          />
        </div>
        <p
          class="pt-3 pb-1 text-sm"
        >
          Product 5
        </p>
        <p
          class="text-sm font-medium"
        >
          500
        </p>
      </a>
    </div>
  </div>
</div>
`;
